code/mob/living/silicon/robot.dm

```diff
@@ -1903,6 +1922,8 @@
 				upgrade.upgrade_activate(src)
 				boutput(src, "[upgrade] has been [upgrade.activated ? "activated" : "deactivated"].")
 		hud.update_upgrades()
+		if (upgrade?.borg_overlay)
+			src.update_appearance()
```

Comment:
This is the bit that actually fixes the bug, the rest is just code formatting. aToMiC pRs.